operator/src/main/java/org/bf2/operator/managers/IngressControllerManager.java

```diff
@@ -537,13 +534,26 @@ private void buildIngressController(String name, String domain,
                 .endSpec();
         }
 
-        ingressReloadIntervalSeconds.ifPresent(value -> {
-            builder.editSpec()
-                    .withNewConfigUnsupportedConfigOverrides()
-                    .withAdditionalProperties(Map.of("reloadInterval", value))
-                    .endConfigUnsupportedConfigOverrides()
-                    .endSpec();
-        });
+        if (hardStopAfter != null && !hardStopAfter.isBlank()) {
+            builder.editMetadata().addToAnnotations(HARD_STOP_AFTER_ANNOTATION, hardStopAfter).endMetadata();
+        } else {
+            builder.editMetadata().removeFromAdditionalProperties(HARD_STOP_AFTER_ANNOTATION).endMetadata();
+        }
+
+
+        // intent here is to preserve any other UnsupportedConfigOverrides and just add/remove reloadInterval as necessary.
+        // Surely there a better way to express this?
+        var specNestedConfigUnsupportedConfigOverridesNested = builder.editSpec().withNewConfigUnsupportedConfigOverrides();
+        HasMetadata current = builder.editSpec().buildUnsupportedConfigOverrides();
+        if (current instanceof Config) {
```

Comment:
Thanks. I see now https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go#L247 and https://github.com/fabric8io/kubernetes-client/blob/296ec4fea3e4369c8a248220fb9abcad969ce6d3/kubernetes-model-generator/openshift-model-operator/src/main/resources/schema/kube-schema.json#L3425 is indeed untypes.

So how is it builder.editSpec().withNewConfigUnsupportedConfigOverrides() has sufficient information to return a nested config builder?  What is it that's hinting to the code generation that that is the required type?  Why the lack of symmetry?


Separately - are there any improvements I can make to the code?

Reply:
> So how is it builder.editSpec().withNewConfigUnsupportedConfigOverrides() has sufficient information to return a nested config builder?

I take back what I was saying about sundrio not knowing about the sub-types.  It does seem to know about the sub-types in the current module (ImagePruner, KubeControllerManager, Storage, etc.) and to things that have a BuildableReference.  So there are withNewXXXUnsupported... methods for all of those - but that does seem kind of arbitrary given how many possible HasMetadata types there are.

> What is it that's hinting to the code generation that that is the required type? Why the lack of symmetry?

It seems like supporting editOrNewUnsupportedConfigOverrides wouldn't be possible as there are multiple types, but it does seem like it could support editOrNewConfigUnsupportedConfigOverrides - and throw a class cast exception if the existing is not a Config.

> Separately - are there any improvements I can make to the code?

To sundrio?

Reply:
> > Separately - are there any improvements I can make to the code?
> 
> To sundrio?

No - I meant to this PR.  Did my last commit leave the code in a reasonable state.

Reply:
> I don't understand what brings withNewConfigUnsupportedConfigOverrides into existance right now.

https://github.com/sundrio/sundrio - it's annotations are added to the generated fabric8 object model https://github.com/fabric8io/kubernetes-client/blob/296ec4fea3e4369c8a248220fb9abcad969ce6d3/kubernetes-model-generator/openshift-model-operator/src/generated/java/io/fabric8/openshift/api/model/operator/v1/IngressControllerSpec.java#L62  It must have logic that checks for all the concrete types - but it is only looking in the current module, or to types it has a BuildableReference to.

> No - I meant to this PR. Did my last commit leave the code in a reasonable state?

Yes, it looks good.